### PR TITLE
Chebyshev port

### LINKED ISSUE #387: port cheyshevU and T to binop_helper
currently `chebyshevU([sym(1) 2 3], x)` probably doesn't work.

Search for other functions that use `binop_helper` to see how to do this.

### PR DESCRIPTION
**Pull request by genuinelucifer**
Fixes #387 

- [x] Port chebyshevU
- [x] Port chebyshevT
- [x] write tests
- [x] write doctests
- [x] Fix all the errors

### CODE REVIEW

**Comment by genuinelucifer** [comment]
@cbm755 Please review. All the tests are passing.
One of the buildbots failed because it could not download some files from ubunut servers (probably a temporary problem not related to this PR).

**Comment by genuinelucifer** [comment]
And I don't know why this many comments are being shown in the PR. I just pulled all the code to master. Then branched from it and did the last 5 commits.

**Comment by cbm755** [comment]
Probably because I rebased (squashed) your previous PR...  Something like `git rebase -i origin/master` to fix this.

To avoid this sort of thing, start a new branch from `origin/master` (or `cbm755/master` or `upstream/master` or whatever you called it).

Let me see if I can rebase this easily...

**Comment by cbm755** [comment]
bit nontrivial to rebase this.  Easiest way might be get a .diff then apply that on a new branch:
`git diff origin/master  > mypatch.diff`.  Do you want to try?  If not, I can do it.

I'll make a few comments as well

File: inst/@sym/chebyshevT.m [hunk 1]
```diff
@@ -20,6 +20,8 @@
 %% @documentencoding UTF-8
 %% @deftypefn  {Function File} {@var{s} =} chebyshevT (@var{n}, @var{x})
 %% Find the nth symbolic Chebyshev polynomial of the first kind.
+%% If n is a vector then it returns a vector with Chebyshev polynomials
+%% of the first kind for each element of 'n'.
```
Comment (cbm755) [inline inst/@sym/chebyshevT.m]: 1.  There needs to be a blank line after the first sentence (for `help chebyshevT`).

2.  This should be written `If @var{n} is a vector then it returns...`

File: inst/@sym/chebyshevT.m [hunk 2]
```diff
@@ -37,6 +39,17 @@
 %% @end group
 %% @end example
 %%
+%% Example with input as vector:
```
Comment (cbm755) [inline inst/@sym/chebyshevT.m]: Actually, I would delete the sentence above and just say:
````
The inputs can be vectors, for example:
````

File: inst/@sym/chebyshevT.m [hunk 3]
```diff
@@ -44,10 +57,10 @@
 %% Keywords: symbolic
 
 function y = chebyshevT(n,x)
-  cmd = { 'n, x = _ins'
-          'return chebyshevt(n, x)' };
+  op = { 'def _op(n, x):'
+        ['    return chebyshevt(n, x)'] };
 
-  y = python_cmd (cmd, sym(n), sym(x));
+  y = binop_helper(n, x, op);
```
Comment (cbm755) [inline inst/@sym/chebyshevT.m]: does `y = binop_helper(n, x, 'chebyshevt')` work?  If so, that's a bit simpler.

**Comment by genuinelucifer** [comment]
Actually i started a new branch only but thus happens. I'll retry.

**Comment by cbm755** [comment]
You may need to add a remote for `cbm755/octsympy`.  E.g., a remote called `upstream`.  Basically, you need a way to keep up with upstream changes...  Then branch off that for new PRs

**Comment by genuinelucifer** [comment]
Hmm. I did that actually. I always keeo origin and upstream. But i guess
this time i messed up something.

**Pull request closed without merging**